src/pybind/rbd/rbd.pyx

```diff
@@ -371,10 +371,10 @@ ELSE:
     cdef rados_ioctx_t convert_ioctx(rados.Ioctx ioctx) except? NULL:
         return <rados_ioctx_t>ioctx.io
 
-cdef int progress_callback(uint64_t offset, uint64_t total, void* ptr) with gil:
+cdef int progress_callback(uint64_t offset, uint64_t total, void* ptr) noexcept with gil:
```

Comment:
Reading https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html:

> ... if you are certain that your function should not raise an exception, (e.g., it does not use Python objects at all, or you plan to use it as a callback in C code that is unaware of Python exceptions), you can declare it as such using `noexcept` or by ...

`ptr` can be an arbitrary Python method/object, passed via `on_progress` parameter.  Does it qualify for the above?

Reply:
By all means, suggest some alternative that will build with Cython 3. As the code stands now it does not build with Cython 3.